pkg/operator/controller/nodeobservability/rbac.go

```diff
@@ -3,163 +3,78 @@ package nodeobservabilitycontroller
-			Labels: labelsForClusterRole(clusterRoleName),
-		},
-		Rules: []rbacv1.PolicyRule{
-			{
-				APIGroups:     []string{secGroup},
-				Resources:     []string{secResource},
-				ResourceNames: []string{secResourceName},
-				Verbs:         []string{use},
-			},
-			{
-				Verbs:     []string{get, list},
-				APIGroups: []string{""},
-				Resources: []string{nodes, nodesProxy, pods},
-			},
-			{
-				Verbs:           []string{get},
-				NonResourceURLs: []string{urlStatus, urlPprof},
-			},
-			{
-				APIGroups: []string{authnGroup},
-				Resources: []string{tokenreviewsResource},
-				Verbs:     []string{create},
-			},
-			{
-				APIGroups: []string{authzGroup},
-				Resources: []string{subjectaccessreviewsResource},
-				Verbs:     []string{create},
-			},
-		},
+		return false, err
 	}
-	return cr
+	return true, nil
 }
 
 // ensureClusterRoleBinding ensures that the clusterrolebinding exists
 // Returns a Boolean value indicating whether it exists, a pointer to the
 // clusterrolebinding and an error when relevant
-func (r *NodeObservabilityReconciler) ensureClusterRoleBinding(ctx context.Context, nodeObs *v1alpha2.NodeObservability, saName, ns string) (bool, *rbacv1.ClusterRoleBinding, error) {
-	nameSpace := types.NamespacedName{Namespace: ns, Name: clusterRoleBindingName}
-	desired := r.desiredClusterRoleBinding(nodeObs, saName, ns)
-	exist, current, err := r.currentClusterRoleBinding(ctx, nameSpace)
-	if err != nil {
-		return false, nil, fmt.Errorf("failed to get ClusterRoleBinding: %v", err)
+func (r *NodeObservabilityReconciler) ensureClusterRoleBinding(ctx context.Context, nodeObs *v1alpha2.NodeObservability, saName, saNamespace string) (*rbacv1.ClusterRoleBinding, error) {
+	desired := r.desiredClusterRoleBinding(nodeObs, saName, saNamespace)
+
+	if err := controllerutil.SetControllerReference(nodeObs, desired, r.Scheme); err != nil {
+		return nil, fmt.Errorf("failed to set the controller reference for clusterrolebindings %s : %w", desired.Name, err)
 	}
-	if !exist {
+
+	current, err := r.currentClusterRoleBinding(ctx)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, fmt.Errorf("failed to get clusterrolebinding %q due to: %w", clusterRoleBindingName, err)
+	} else if err != nil && errors.IsNotFound(err) {
+
+		// create clusterrolebinding since it is not found
 		if err := r.createClusterRoleBinding(ctx, desired); err != nil {
-			return false, nil, err
+			return nil, fmt.Errorf("failed to create clusterrolebinding %q: %w", clusterRoleBindingName, err)
 		}
-		return r.currentClusterRoleBinding(ctx, nameSpace)
+
+		r.Log.Info("created clusterrolebinding", "clusterrolebinding.Name", clusterRoleBindingName)
+		return r.currentClusterRoleBinding(ctx)
 	}
-	return true, current, err
+
+	if _, err = r.updateClusterRoleBinding(ctx, current, desired); err != nil {
+		return nil, fmt.Errorf("failed to update clusterrolebinding %s due to: %w", clusterRoleBindingName, err)
+	}
+
+	return r.currentClusterRoleBinding(ctx)
```

Comment:
Can you please do the same as in the daemonset controller?

- return `true` if the clusterrolebinding was updated and call `currentClusterRoleBinding` and log the `updated` message
- return `current` variable otherwise